mlx_graphs/utils/validators.py

```diff
@@ -65,13 +65,18 @@ def validate_list_of_graph_data(func):
 
     @functools.wraps(func)
     def wrapper(graph_list: list[GraphData], *args, **kwargs):
-        expected_attr = set(graph_list[0].to_dict())
+        if not isinstance(graph_list, list):
+            raise ValueError(f"Expected list of GraphData, got {type(graph_list)}.")
+
         for i, graph in enumerate(graph_list):
             if not isinstance(graph, GraphData):
                 raise ValueError(
+                    "Expected list of GraphData. "
                     f"Graph at index {i} in the batch is not of type `GraphData`."
                 )
 
+        expected_attr = set(graph_list[0].to_dict())
+        for i, graph in enumerate(graph_list):
```

Comment:
Why doing the same loop twice?

Reply:
because if the elements are not `GraphData`, `graph_list[0]` don't have the `to_dict` method. Anyways, you're right it's not efficient so I changed how `expected_attr` is handled